ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/utils/ThreadUtil.java

```diff
@@ -28,7 +31,7 @@
     /**
      * The ID associated with the thread.
      */
-    private static final ThreadLocal THREAD_ID = new ThreadLocal() ;
+    private static final Map<Thread,String> THREAD_ID = new HashMap<Thread,String>() ;
```

Comment:
well, I can acknowledge that but if I moved to using a long for threadId then we would end up changing the public interface of some classes. Initially I found BasicAction::removeChildThread(String threadId) - there could be more....